app/containers/ExplorePage.tsx

```diff
@@ -51,45 +52,110 @@ function chooseFile(
+
+  const filterCondition = (needle, haystack) => {
+    if (haystack.length === 0) return true;
+    return haystack.map(entry => entry.value).includes(needle);
+  };
+
+  const getFilteredData = options => {
+    let filtered = options.observations;
+
+    if (typeof filters !== 'undefined') {
+      filtered = filtered.filter(entry => {
+        const validEntry =
+          filterCondition(entry.pred_1, filters.activeAnimals) &&
+          filterCondition(entry.camera, filters.activeCameras) &&
+          filterCondition(entry.station, filters.activeStations) &&
+          filterCondition(entry.check, filters.activeChecks);
+        return validEntry;
+      });
+    }
+    return { observations: filtered };
+  };
+
+  const MainPanel: React.SFC = () => (
+    <div style={{ display: 'flex' }}>
+      <div
+        style={{
+          flex: '1',
+          width: '55%',
+          paddingBottom: '20px',
+          marginRight: '10px'
+        }}
+      >
+        <Card style={{ height: '100%' }} interactive elevation={Elevation.TWO}>
+          <AnimalsPlot data={getFilteredData(data)} />
+        </Card>
+      </div>
+      <div
+        style={{
+          flex: '1',
+          width: '45%',
+          paddingBottom: '20px',
+          marginLeft: '10px'
+        }}
+      >
+        <Card style={{ height: '100%' }} interactive elevation={Elevation.TWO}>
+          <Map data={getFilteredData(data)} />
+        </Card>
+      </div>
+    </div>
+  );
+  const TablePanel: React.SFC = () => (
+    <ObservationsTable style={{ width: '100%' }} data={getFilteredData(data)} />
+  );
 
   const contents =
     data !== undefined ? (
       <>
-        <h1>{t('Explore')}</h1>
-        <Button
-          text={t('Back')}
-          icon="arrow-left"
-          onClick={() => setData(undefined)}
-          style={{ marginBottom: '10px', backgroundColor: '#fff' }}
-        />
-        ,
-        {/* <div style={{ flex: 1, paddingBottom: '20px' }}>
-          <Card interactive elevation={Elevation.TWO}>
-            <h2 style={{ marginTop: 0 }}>{t('Filters')}</h2>
-          </Card>
-        </div> */}
-        ,
-        <div style={{ flex: 1, width: '100%', paddingBottom: '20px' }}>
-          <Card interactive elevation={Elevation.TWO}>
-            <AnimalsPlot data={data} />
-          </Card>
+        <div>
+          <h1>{t('Explore')}</h1>
+        </div>
+        <ExplorerFilter data={data} onChange={handleFilters} />
+        <div style={{ minHeight: '30px' }}>
+          <span>
+            <span style={{ fontWeight: 600 }}>{t('Number of results')}</span>
+            <span style={{ paddingLeft: '5px' }}>
+              {getFilteredData(data).observations.length}
+            </span>
+          </span>
         </div>
-        ,
-        <div style={{ flex: 1, width: '100%', paddingBottom: '20px' }}>
-          <Card interactive elevation={Elevation.TWO}>
-            <Map data={data} />
-          </Card>
+        <div>
+          <Button
```

Comment:
this button is in a strange place (i know i put it here... it was bad from the beginning now it's even more counterintuitive), how should we provide the ability to go back and select a new file for analysis? 



some other solution than a "Back" button (like [breadcrumbs from blueprint](https://blueprintjs.com/docs/#core/components/breadcrumbs) or maybe simply move it somewhere else and possibly change its label?

Reply:
Improved layout